addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -240,14 +245,30 @@ def cancel_ewaybill(self):
         }
 
     def reset_to_pending(self):
-        if self.state != 'cancel':
-            raise UserError(_("Only Cancelled E-waybill can be resent."))
+        self.ensure_one()
+        if self.state not in ('cancel', 'challan'):
+            raise UserError(_("Only Delivery Challan and Cancelled E-waybill can be reset to pending."))
         self.write({
             'state': 'pending',
             'cancel_reason': False,
             'cancel_remarks': False,
         })
 
+    def set_to_challan(self):
+        self.ensure_one()
+        if self.state in ['generated', 'cancel']:
+            raise UserError(_(
+                "The e-Waybill (%s) cannot be directly converted into a challan. "
+                "Please cancel the e-Waybill and reset it to pending in order to create a "
+                "challan.",
+                self.name
+            ))
+        elif self.state == 'challan':
+            raise UserError(_("The record is already in the challan state."))
```

Comment:
Should be taken care if mostly by the view
If state != 'pending':
    raise UserError(..
would already be too much